@navikt/core/react/src/form/combobox/SelectedOptions/selectedOptionsContext.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import usePrevious from "../../../util/usePrevious";
+import { useInputContext } from "../Input/inputContext";
+import { ComboboxProps } from "../types";
+
+type SelectedOptionsContextType = {
+  selectedOptions: string[];
+  addSelectedOption: (option: string) => void;
+  removeSelectedOption: (option: string) => void;
+  setSelectedOptions: (any) => void;
+  prevSelectedOptions?: string[];
+  singleSelect?: boolean;
+};
+
+const SelectedOptionsContext = createContext<SelectedOptionsContextType>(
+  {} as SelectedOptionsContextType
+);
+
+export const SelectedOptionsProvider = ({
+  children,
+  value,
+}: {
+  children: any;
+  value: Pick<
+    ComboboxProps,
+    "selectedOptions" | "singleSelect" | "onToggleSelected"
+  >;
+}) => {
+  const { setSearchTerm, setValue } = useInputContext();
+  const {
+    selectedOptions: externalSelectedOptions,
+    singleSelect,
+    onToggleSelected,
+  } = value;
+  const [selectedOptions, setSelectedOptions] = useState<string[]>([]);
+
+  useEffect(
+    () => setSelectedOptions(externalSelectedOptions || []),
+    [externalSelectedOptions]
+  );
```

Comment:
```suggestion
if(selectedOptions !== externalSelectedOptions){
   setSelectedOptions(externalSelectedOptions || [])
}
```
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
Ifølge React vil dette være den best måten å lytte til eksterne props. Virker litt skummelt å ha rene if-tester med setState rett i koden, men er det react ønsker 🤷

Reply:
Eksempelet til React har et ekstra element som ser viktig ut her, nemlig prevItem/setPrevItem.
Uten den får vi ikke gjort en sjekk på om externalSelectedOptions har endret seg.

Men, de lister jo opp det de mener er en enda bedre løsning, nemlig å bare "kalkulere verdien under rendering".
Det ser ut til å fungere hos oss, med følgende kode:

`const [internalSelectedOptions, setSelectedOptions] = useState<string[]>([]);
 const selectedOptions = externalSelectedOptions ?? internalSelectedOptions;`
 
 Vurderte en useMemo i stedet, men her er det jo bare snakk om å velge hvilken verdi man skal bruke. Enten bruker vi den eksterne, eller så bruker vi den interne. Det er ikke en tung kalkulasjon.
 
 Satt litt fast på at det ikke fungerte, frem til jeg innså at storybook-eksemplene sender jo inn en verdi for selectedOptions. Den ble egentlig brukt som "intitalSelectedOptions", så det blir en liten funksjonell endring her.
 Jeg tror det er greit (sender man inn selectedOptions, så må man også håndtere oppdateringer av den), men vi kan evt. vurderere en "defaultSelectedOptions" a la "defaultValue" om noen spør etter det.

Reply:
🙌 Hvis vi følger pattern som er brukt i de fleste andre komponenter gir `defaultSelectedOptions` mye mening. Hvis de setter `selectedOptions` selv må de også ta seg av oppdateringen som du nevner 👍